fabtests/prov/efa/src/rdm_rnr_queue_resend.c

```diff
@@ -110,10 +112,15 @@ static int alloc_atomic_res(struct fi_info *fi, void **result, void **compare,
 			0, 0, mr_compare, NULL);
 	if (ret) {
 		FT_PRINTERR("fi_mr_reg", ret);
-		return ret;
+		goto err3;
```

Comment:
Not blocking this PR but not sure why Coverity didn't flag this.
Need to unregister mr_result if register mr_compare fails.
Just good practice for a function to clean up after itself on failure.

Reply:
@chien-intel I generally agree but I think fabtests has a convention to never clean up fids/OFI resources in the moment and always clean them up when the parent return happens, closing all the fids at once. I think this can follow the same pattern and let the cleanup handle the fid closing if needed.
That said, I think this test doesn't actually call cleanup properly :) It should be calling free_res on error and add the atomic close call in there as well. ie
this
```
	ret = rnr_queue_resend_test(req_pkt, atomic_op);
	if (ret) {
		FT_PRINTERR("rnr_queue_resend_test", -ret);
		return ret;
	}

	ret = ft_close_oob();
	if (ret) {
		FT_PRINTERR("ft_close_oob", -ret);
		return ret;
	}
	ft_free_res();

	return 0;
```
should be something like
```

static int run(int req_pkt, enum fi_op atomic_op)
{
	int ret;

	ret = ft_efa_rnr_init_fabric();
	if (ret) {
		FT_PRINTERR("ft_efa_rnr_init_fabric", -ret);
		goto out;
	}

	ret = rnr_queue_resend_test(req_pkt, atomic_op);
	if (ret) {
		FT_PRINTERR("rnr_queue_resend_test", -ret);
		goto out;
	}

	ret = ft_close_oob();
	if (ret) {
		FT_PRINTERR("ft_close_oob", -ret);
		return ret;
	}

out:
	free_atomic_res();
	ft_free_res();

	return 0;
}
```
But the atomic resources go out of scope returning from the resend test. Really, I think this test needs some refactoring to handle cleanup better. @shijin-aws @a-szegel Can anyone from AWS handle this cleanup so we don't mess up your test too much?

Reply:
> @chien-intel I generally agree but I think fabtests has a convention to never clean up fids/OFI resources in the moment and always clean them up when the parent return happens, closing all the fids at once.

It is best I stop commenting here. :-)

Reply:
I agree with @aingerson, I will create a separate PR to refactor and handle this.

Reply:
Opened https://github.com/ofiwg/libfabric/pull/9848, you can drop this change in this PR

Reply:
change dropped